libcore/File.vala

```diff
@@ -0,0 +1,1177 @@
+public class GOF.File : GLib.Object {
+    private static GLib.HashTable<GLib.File, GOF.File> file_cache;
+
+    public enum IconFlags {
+        NONE,
+        USE_THUMBNAILS
+    }
+
+    public enum ThumbState {
+        UNKNOWN = 0x00,
+        NONE = 0x01,
+        READY = 0x02,
+        LOADING = 0x03
+    }
+
+    public const string GIO_DEFAULT_ATTRIBUTES = "standard::is-hidden,standard::is-backup,standard::is-symlink,standard::type,standard::name,standard::display-name,standard::content-type,standard::fast-content-type,standard::size,standard::symlink-target,standard::target-uri,access::*,time::*,owner::*,trash::*,unix::*,id::filesystem,thumbnail::*,mountable::*,metadata::marlin-sort-column-id,metadata::marlin-sort-reversed";
+
+    public signal void changed ();
+    public signal void icon_changed ();
+    public signal void destroy ();
+
+    public bool is_gone;
+    public GLib.File location = null;
+    public GLib.File target_location = null;
+    public GOF.File target_gof = null;
+    public GLib.File directory = null; /* parent directory location */
```

Comment:
Can we use a property here returning [GLib.File's get_parent () method](https://valadoc.org/gio-2.0/GLib.File.get_parent.htm)?

Reply:
Yes, but I think the idea of GOF.File is to speed things up by caching a lot of file info.  GOF.FIle.directory is assigned once on creation.